Torch tests fail.

## 🐛 Bug report

By default the developer environment does not install torch, so it is not tested. If the latest version of torch is installed (`2.4.1+cu121`) there are 27 failing tests. 

```
class_name = 'PolynomialLR'

    @pytest.mark.skipif(not torch_available, reason="torch>=2.1 package is required")
    @pytest.mark.parametrize(
        "class_name",
        [
            "_LRScheduler",
            "LambdaLR",
            "MultiplicativeLR",
            "StepLR",
            "MultiStepLR",
            "ConstantLR",
            "LinearLR",
            "ExponentialLR",
            "ChainedScheduler",
            "SequentialLR",
            "CosineAnnealingLR",
            "ReduceLROnPlateau",
            "CyclicLR",
            "CosineAnnealingWarmRestarts",
            "OneCycleLR",
            "PolynomialLR",
        ],
    )
    def test_get_params_torch_lr_scheduler(class_name):
        cls = getattr(torch.optim.lr_scheduler, class_name)
        params = get_params(cls)
>       assert all(p.annotation is not inspect._empty for p in params)
E       assert False
E        +  where False = all(<generator object test_get_params_torch_lr_scheduler.<locals>.<genexpr> at 0x743acdfe79f0>)

jsonargparse_tests/test_stubs_resolver.py:307: AssertionError
================================================================================= warnings summary =================================================================================
venv311/lib/python3.11/site-packages/torch/_subclasses/functional_tensor.py:258
  /root/repo/venv311/lib/python3.11/site-packages/torch/_subclasses/functional_tensor.py:258: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:84.)
    cpu = _conversion_method_template(device=torch.device("cpu"))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= short test summary info ==============================================================================
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[Adadelta] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[Adagrad] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[Adamax] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[ASGD] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[LBFGS] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[NAdam] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[RAdam] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[RMSprop] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[Rprop] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[SGD] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_optimizer[SparseAdam] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[_LRScheduler] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[LambdaLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[MultiplicativeLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[StepLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[MultiStepLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[ConstantLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[LinearLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[ExponentialLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[ChainedScheduler] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[SequentialLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[CosineAnnealingLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[ReduceLROnPlateau] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[CyclicLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[CosineAnnealingWarmRestarts] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[OneCycleLR] - assert False
FAILED jsonargparse_tests/test_stubs_resolver.py::test_get_params_torch_lr_scheduler[PolynomialLR] - assert False
```

However, on torch `2.1.1`, `2.3.0`, and `2.3.1` it does work. It starts to fail on `2.4.0`.

### Environment

<!-- Fill in the list below. -->

- jsonargparse version: main - 88c038745f68bfd56d830ad58ed42e2b1035440d (post v4.32.1)
- Python version: 3.11
- How jsonargparse was installed: pip developer install
- OS: Ubuntu 22.04

--- a/jsonargparse_tests/test_stubs_resolver.py
+++ b/jsonargparse_tests/test_stubs_resolver.py
@@ -246,14 +246,18 @@ def test_get_params_complex_function_requests_get(parser):
 
 
 if torch_available:
-    import torch.optim  # pylint: disable=import-error
-    import torch.optim.lr_scheduler  # pylint: disable=import-error
+    import pkg_resources
 
-    if tuple(int(v) for v in torch.__version__.split(".", 2)[:2]) < (2, 1):
+    torch_version = tuple(int(v) for v in pkg_resources.get_distribution("torch").version.split(".", 2)[:2])
+
+    if torch_version < (2, 1) or torch_version >= (2, 4):
         torch_available = False
+    else:
+        import torch.optim  # pylint: disable=import-error
+        import torch.optim.lr_scheduler  # pylint: disable=import-error
 
 
-@pytest.mark.skipif(not torch_available, reason="torch>=2.1 package is required")
+@pytest.mark.skipif(not torch_available, reason="only for torch>=2.1,<2.4")
 @pytest.mark.parametrize(
     "class_name",
     [
@@ -279,7 +283,7 @@ def test_get_params_torch_optimizer(class_name):
     assert any(p.annotation is inspect._empty for p in params)
 
 
-@pytest.mark.skipif(not torch_available, reason="torch>=2.1 package is required")
+@pytest.mark.skipif(not torch_available, reason="only for torch>=2.1,<2.4")
 @pytest.mark.parametrize(
     "class_name",
     [

